lib/apirequest.js

```diff
@@ -184,6 +186,10 @@ function createAPIRequest(parameters, callback) {
   delete options.auth; // is overridden by our auth code
   delete options.params; // We handle params ourselves and Request does not recognise 'params'
 
+  if (missingCallback) {
```

Comment:
You're right. If no callback is given then it's no longer an error so we can get rid of createCallback and logError function in this file.